### PR TITLE
Implement a basic URL scheme.

### LINKED ISSUE #307: Implement a registered Heroic protocol.
For integration purposes, we need to implement a standardised protocol for communicating with Heroic from other applications. The simplest way to achieve this is using Electron's [protocol](https://www.electronjs.org/docs/api/protocol). 

Candidates:
- `heroic://` (going with this one.)
- `heroicgl://`

I'm not sure which kind of protocol we should register, but I'm thinking string.

Usecases:
- Desktop Shortcuts
- Integration into other services like Discord Rich Presence. (I'm thinking: Open Game button to the Rich Presence so that anybody can play along.)

### PR DESCRIPTION
**Pull request by adityaruplaha**
Fixes #307.

Implementation based on what I read here:
https://github.com/electron-userland/electron-builder/issues/4035
https://github.com/electron/electron/issues/4857

Scheme is a lot like HTTP, except that there's no domain name.
Like `heroic://launch?appName=Hazelnut`.

### CODE REVIEW

**Comment by flavioislima** [comment]
Maybe to test you just need to put
`open heroic://`
on the terminal
or xdg-open, like we are doing with the urls to open the epic login page.

**Comment by adityaruplaha** [comment]
Done. Should work perfectly.

File: electron/main.ts [hunk 1]
```diff
@@ -204,6 +206,21 @@ if (!gotTheLock) {
       ]
     })
 
+    protocol.registerStringProtocol('heroic', (request, callback) => {
+      const [command, args_string] = request.url.split('://')[1].split('/')
+      handleProtocol(command, args_string.split('::'))
+      callback('Operation initaited.')
```
Comment (flavioislima) [inline electron/main.ts:212]: just a typo

**Comment by adityaruplaha** [comment]
@flavioislima this is ready to be merged, but problem is that the frontend has no idea what just happened. I'll make the backend state-aware and the client can just query it. Would also reduce the global state.

File: electron/protocol.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+          'defaultPath': path,
+          'message': 'Select Installation Path',
+          'properties': ['createDirectory', 'openDirectory']
+        })
+        path = filePaths[0] || path
+      }
+      return await game.install(path)
+    }
+  }
+}
```
Comment (adityaruplaha) [inline electron/protocol.ts:69]: Whoopsie.

**Review by flavioislima — APPROVED** [review]
Looks good. I'm just not 100% on the user confirmation because if the desktop file will be clicked, then I think it was on porpuse. Steam and lutris don't ask for confirmation for instance. So I think this might not be necessary. But merge it and we can think about that before release.

**Comment by adityaruplaha** [comment]
> I'm just not 100% on the user confirmation

I'm not either, but I did it just in case as its easier to remove something than to implement it pre-release.

File: electron/protocol.ts (hunk 2 above)
Comment (adityaruplaha) [reply electron/protocol.ts:69]: Fixed by #314.

**Pull request merged**